sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/httpurlconnection/HttpUrlConnectionClientBuilder.java

```diff
@@ -160,11 +135,24 @@ public HttpClient build() {
             ? Configuration.getGlobalConfiguration()
             : configuration;
 
+        ProxyOptions buildProxyOptions = (proxyOptions == null)
+            ? ProxyOptions.fromConfiguration(buildConfiguration)
+            : proxyOptions;
+
+        if (buildProxyOptions != null && buildProxyOptions.getUsername() != null) {
+            Authenticator.setDefault(new ProxyAuthenticator(buildProxyOptions.getUsername(),
+                buildProxyOptions.getPassword()));
+        }
+
+        if (buildProxyOptions != null && buildProxyOptions.getType() != ProxyOptions.Type.HTTP
+            && buildProxyOptions.getType() != null) {
+            throw new IllegalArgumentException("Invalid proxy");
+        }
+
         return new HttpUrlConnectionClient(
             getTimeout(connectionTimeout, DEFAULT_CONNECT_TIMEOUT),
             getTimeout(readTimeout, DEFAULT_READ_TIMEOUT),
-            getTimeout(writeTimeout, DEFAULT_WRITE_TIMEOUT),
-            getTimeout(responseTimeout, DEFAULT_RESPONSE_TIMEOUT));
+            buildProxyOptions);
```

Comment:
If we removed a couple timeout attributes in favor of using `ProxyOptions`, maybe we should put all of them there.

Reply:
The attributes that were removed were not relevant for HUC , I think they were there from our other clients supported contexts in core.